src/library/trackset/baseplaylistfeature.cpp

```diff
@@ -139,44 +145,70 @@ void BasePlaylistFeature::initActions() {
 int BasePlaylistFeature::playlistIdFromIndex(const QModelIndex& index) {
     TreeItem* item = static_cast<TreeItem*>(index.internalPointer());
     if (item == nullptr) {
-        return -1;
+        return kInvalidPlaylistId;
     }
 
     bool ok = false;
     int playlistId = item->getData().toInt(&ok);
     if (ok) {
         return playlistId;
     } else {
-        return -1;
+        return kInvalidPlaylistId;
+    }
+}
+
+void BasePlaylistFeature::selectPlaylistInSidebar(int playlistId) {
+    if (!m_pSidebarWidget) {
+        return;
+    }
+    if (playlistId == kInvalidPlaylistId) {
+        return;
+    }
+    QModelIndex index = indexFromPlaylistId(playlistId);
+    if (index.isValid() && m_pSidebarWidget) {
+        m_pSidebarWidget->selectChildIndex(index);
     }
 }
 
 void BasePlaylistFeature::activateChild(const QModelIndex& index) {
     //qDebug() << "BasePlaylistFeature::activateChild()" << index;
     int playlistId = playlistIdFromIndex(index);
-    if (playlistId != -1) {
-        m_pPlaylistTableModel->setTableModel(playlistId);
-        emit showTrackModel(m_pPlaylistTableModel);
-        emit enableCoverArtDisplay(true);
+    if (playlistId == kInvalidPlaylistId) {
+        return;
+    }
+
+    m_pPlaylistTableModel->setTableModel(playlistId);
+    emit showTrackModel(m_pPlaylistTableModel);
+    emit enableCoverArtDisplay(true);
+
+    if (!m_pSidebarWidget) {
+        return;
     }
+    m_pSidebarWidget->selectChildIndex(index);
 }
 
 void BasePlaylistFeature::activatePlaylist(int playlistId) {
-    //qDebug() << "BasePlaylistFeature::activatePlaylist()" << playlistId;
+    // qDebug() << "BasePlaylistFeature::activatePlaylist()" << playlistId;
+    if (playlistId == kInvalidPlaylistId) {
+        return;
+    }
     QModelIndex index = indexFromPlaylistId(playlistId);
-    if (playlistId != -1 && index.isValid()) {
-        m_pPlaylistTableModel->setTableModel(playlistId);
-        emit showTrackModel(m_pPlaylistTableModel);
-        emit enableCoverArtDisplay(true);
-        // Update selection
-        emit featureSelect(this, m_lastRightClickedIndex);
-        activateChild(m_lastRightClickedIndex);
+    if (!index.isValid()) {
+        return;
     }
+
+    m_lastRightClickedIndex = index;
+    m_pPlaylistTableModel->setTableModel(playlistId);
+    emit showTrackModel(m_pPlaylistTableModel);
+    emit enableCoverArtDisplay(true);
+    // Update selection
+    emit featureSelect(this, m_lastRightClickedIndex);
+    activateChild(m_lastRightClickedIndex);
```

Comment:
```suggestion
    if (!m_pSidebarWidget) {
        m_pSidebarWidget->selectChildIndex(index);
    }
```

is this sufficient here?